definitions/output/reports/tech_crux.js

```diff
@@ -0,0 +1,297 @@
+    tech.technology != '' AND
+    REGEXP_EXTRACT_ALL(version, r'(0|[1-9]\\d*)\\.(?:0|[1-9]\\d*)\\.(?:0|[1-9]\\d*)')[SAFE_OFFSET(0)] IS NOT NULL
+
+  UNION ALL
+
+  SELECT
+    tech.technology,
+    'ALL' AS version,
+    client,
+    page
+  FROM pages,
+    UNNEST(technologies) AS tech
+
+  UNION ALL
+
+  SELECT
+    'ALL' AS technology,
+    'ALL' AS version,
+    client,
+    page
+  FROM pages
+),
+
+categories AS (
+  SELECT
+    tech.technology,
+    ARRAY_TO_STRING(ARRAY_AGG(DISTINCT category IGNORE NULLS ORDER BY category), ', ') AS category
+  FROM pages,
+    UNNEST(technologies) AS tech,
+    UNNEST(tech.categories) AS category
+  GROUP BY technology
+
+  UNION ALL
+
+  SELECT
+    'ALL' AS technology,
+    ARRAY_TO_STRING(ARRAY_AGG(DISTINCT category IGNORE NULLS ORDER BY category), ', ') AS category
+  FROM pages,
+    UNNEST(technologies) AS tech,
+    UNNEST(tech.categories) AS category
+  WHERE
+    client = 'mobile'
+),
+
+lab_metrics AS (
+  SELECT
+    client,
+    page,
+    origin,
+    SAFE.INT64(summary.bytesTotal) AS bytesTotal,
+    SAFE.INT64(summary.bytesJS) AS bytesJS,
+    SAFE.INT64(summary.bytesImg) AS bytesImg,
+    SAFE.FLOAT64(lighthouse.categories.accessibility.score) AS accessibility,
+    SAFE.FLOAT64(lighthouse.categories['best-practices'].score) AS best_practices,
+    SAFE.FLOAT64(lighthouse.categories.performance.score) AS performance,
+    SAFE.FLOAT64(lighthouse.categories.pwa.score) AS pwa,
+    SAFE.FLOAT64(lighthouse.categories.seo.score) AS seo
+  FROM pages
+),
+
+lab_data AS (
+  SELECT
+    client,
+    origin,
+    technology,
+    version,
+    ANY_VALUE(category) AS category,
+    AVG(bytesTotal) AS bytesTotal,
+    AVG(bytesJS) AS bytesJS,
+    AVG(bytesImg) AS bytesImg,
+    AVG(accessibility) AS accessibility,
+    AVG(best_practices) AS best_practices,
+    AVG(performance) AS performance,
+    AVG(pwa) AS pwa,
+    AVG(seo) AS seo
+  FROM lab_metrics
+  INNER JOIN technologies
+  USING (client, page)
+  INNER JOIN categories
+  USING (technology)
+  GROUP BY
+    client,
+    origin,
+    technology,
+    version
+)
+
+SELECT
+  DATE('${pastMonth}') AS date,
+  geo,
+  client,
+  rank,
+  technology,
+  version,
+  COUNT(DISTINCT origin) AS origins,
+
+  # CrUX data
+  COUNTIF(good_fid) AS origins_with_good_fid,
+  COUNTIF(good_cls) AS origins_with_good_cls,
+  COUNTIF(good_lcp) AS origins_with_good_lcp,
+  COUNTIF(good_fcp) AS origins_with_good_fcp,
+  COUNTIF(good_ttfb) AS origins_with_good_ttfb,
+  COUNTIF(good_inp) AS origins_with_good_inp,
+  COUNTIF(any_fid) AS origins_with_any_fid,
+  COUNTIF(any_cls) AS origins_with_any_cls,
+  COUNTIF(any_lcp) AS origins_with_any_lcp,
+  COUNTIF(any_fcp) AS origins_with_any_fcp,
+  COUNTIF(any_ttfb) AS origins_with_any_ttfb,
+  COUNTIF(any_inp) AS origins_with_any_inp,
+  COUNTIF(good_cwv_2024) AS origins_with_good_cwv,
+  COUNTIF(good_cwv_2024) AS origins_with_good_cwv_2024,
+  COUNTIF(good_cwv_2023) AS origins_with_good_cwv_2023,
```

Comment:
@rviscomi as I'll be recalculating the historical data, should I merge these into a single column based on date condition?

Reply:
`origins_with_good_cwv` should definitely have that logic. Whether we need the other two once we have that is debatable.

Reply:
Didn't get it.
I thought to deduplicate in the final SELECT:
```suggestion
  COUNTIF(IF(date < '2024-01-01', good_cwv_2023, good_cwv_2024)) AS origins_with_good_cwv,
```
WDYT?

Reply:
Yes if doing that, it’s fine. But at the moment you’re returning 3 fields - the first of which is incorrect AND just a duplicate of good 2024.

Reply:
Ok, if nobody using those, will deduplicate.